java/src/main/java/com/anaconda/skein/MsgUtils.java

```diff
@@ -609,13 +607,22 @@ public static Level readLogLevel(Msg.Log.Level logLevel) {
       fileSystems.add(new Path(spec.getFileSystems(i)));
     }
 
+    final List<CredentialProvider> credentialProviders = new LinkedList<CredentialProvider>();
+    for(int i = 0; i < spec.getCredentialProvidersCount(); i++) {
+      Msg.CredentialProviderSpec d = spec.getCredentialProviders(i);
+      if (d.getName().equals("hive")) {
+        credentialProviders.add(new HiveCredentials(d, spec.getUser()));
```

Comment:
What should we do if the user requests a provider we don't support? There's two cases here:
- The name provided doesn't match any of our implementations.
- The name does match one we implement, but adding a credential fails.

In both cases I think we should error. But I don't think that should be handled here, rather I think it should be handled in the routine that manages adding credentials (described below).

Reply:
yes, there will be exceptions propagated backwards in both cases in the next draft